cypress-tests/cypress/support/pageBuilder/pbDeleteAllCategories.ts

```diff
@@ -0,0 +1,70 @@
+import { GraphQLClient } from "graphql-request";
+
+declare global {
+    // eslint-disable-next-line @typescript-eslint/no-namespace
+    namespace Cypress {
+        interface Chainable {
+            pbDeleteAllCategories(): void;
+        }
+    }
+}
+
+const LIST_CATEGORIES_QUERY = /* GraphQL */ `
+    query ListCategories {
+        pageBuilder {
+            listCategories {
+                data {
+                    slug
+                }
+            }
+        }
+    }
+`;
+
+const DELETE_CATEGORY_MUTATION = /* GraphQL */ `
+    mutation DeleteCategory($slug: String!) {
+        pageBuilder {
+            deleteCategory(slug: $slug) {
+                error {
+                    code
+                    message
+                }
+            }
+        }
+    }
+`;
+
+Cypress.Commands.add("pbDeleteAllCategories", () => {
+    cy.login().then(user => {
+        const client = new GraphQLClient(Cypress.env("GRAPHQL_API_URL"), {
+            headers: {
+                authorization: `Bearer ${user.idToken.jwtToken}`
+            }
+        });
+
+        // Step 1: Fetch categories
+        client.request(LIST_CATEGORIES_QUERY).then(response => {
+            const categories = response.pageBuilder.listCategories.data;
+
+            // Step 2: Filter and delete categories
+            categories.forEach(category => {
```

Comment:
This won't work because `forEach` is not async. 

We've been in this situation before. When dealing with promises, we want to return them from functions because only if Cypress receives a promise, it can know when to proceed with next step.